createdisk.sh

```diff
@@ -58,13 +58,19 @@ function create_qemu_image {
             exit 1
     fi
 
+    # Remove immutable attribute for /dev/sda4
+    guestfish -a $destDir/${CRC_VM_NAME}.qcow2 -m /dev/sda4 set-e2attrs / i clear:true
+
     # TMPDIR must point at a directory with as much free space as the size of the image we want to sparsify
     # Read limitation section of `man virt-sparsify`.
     TMPDIR=$(pwd)/$destDir ${VIRT_SPARSIFY} -o lazy_refcounts=on $destDir/${CRC_VM_NAME}.qcow2 $destDir/${CRC_VM_NAME}_sparse.qcow2
     rm -f $destDir/${CRC_VM_NAME}.qcow2
     mv $destDir/${CRC_VM_NAME}_sparse.qcow2 $destDir/${CRC_VM_NAME}.qcow2
     rm -fr $destDir/.guestfs-*
 
+    # Set immutable attribute back for /dev/sda4
+    guestfish -a $destDir/${CRC_VM_NAME}.qcow2 -m /dev/sda4 set-e2attrs / i
+
```

Comment:
Won't that cause issues with 4.2 which does not set this bit?

Reply:
@cfergeau I didn't test that but it shouldn't if we just want to make that partition as immutable, Need to test this out and if this is causing issue we might just create another branch for 4.2 side.

Reply:
On 4.2 the partition is not immutable, so I don't know how coreos is going to behave if we suddenly make it immutable. We could check whether the partition is immutable or not before making these changes.

Reply:
@cfergeau on 4.2 side there is no `/dev/sda4` :(

```
$ guestfish -a crc.qcow2 -m /dev/sda4 set-e2attrs / i
libguestfs: error: vfs_type: vfs_type_stub: /dev/sda4: No such file or directory
libguestfs: error: mount_options: mount_options_stub: /dev/sda4: No such file or directory
guestfish: ‘/dev/sda4’ could not be mounted.
guestfish: Did you mean to mount one of these filesystems?
guestfish: 	/dev/sda1 (unknown)
guestfish: 	/dev/sda2 (ext4)
guestfish: 	/dev/sda3 (xfs)
```

Reply:
> On 4.2 the partition is not immutable, so I don't know how coreos is going to behave if we suddenly make it immutable. We could check whether the partition is immutable or not before making these changes.

@cfergeau yes, that would be a better solution in terms of checks. Using the guestfish we can get if there is immutable bit set and then only perform removing that bit and adding it but as I said for the 4.2 side there is no `/dev/sda4`.

```
$ guestfish -a crc.qcow2 -m /dev/sda4 get-e2attrs /
i
```

Reply:
@cfergeau 

For 4.3 images
```
guestfish <<_EOF_
 add crc.qcow2
run
list-filesystems
_EOF_
/dev/sda1: ext4
/dev/sda2: vfat
/dev/sda3: unknown
/dev/sda4: xfs
```

For 4.2 images
```
$  guestfish <<_EOF_
>  add crc.qcow2
> run
> list-filesystems
> _EOF_
/dev/sda1: unknown
/dev/sda2: ext4
/dev/sda3: xfs
```

Reply:
Oh well, if no /dev/sda4 on 4.2, this indeed means my concern is not going to be an issue. We could leave it as it is and ignore the error.
Looks like we should double check we resize the correct partition on 4.3?

Reply:
> Looks like we should double check we resize the correct partition on 4.3?

Yes now it should be `/dev/sda4` which we need to resize `/dev/sda3` is the boot partition so I think first we detect which partition is `xfs` and then increase it size.

```
$ sudo fdisk -l
Disk /dev/vda: 31 GiB, 33285996544 bytes, 65011712 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 00000000-0000-4000-A000-000000000001

Device        Start      End  Sectors  Size Type
/dev/vda1      2048   788479   786432  384M Linux filesystem
/dev/vda2    788480  1048575   260096  127M EFI System
/dev/vda3   1048576 32505215 31456640   15G BIOS boot
/dev/vda4  32505216 65008990 32503775 15.5G Linux filesystem
```